src/main/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandler.java

```diff
@@ -154,21 +154,48 @@ public PaymentHandleResponse createPayment(@Nonnull String ctpPaymentId) {
+                            } else {
+                                return PaymentHandleResponse.of400BadRequest(
+                                        format("Payment payment=[%s] does not exists or does not have a connected cart.",
+                                                ctpPaymentId));
+                            }
+                        })
+                .toCompletableFuture().join();
+    }
+
     public PaymentHandleResponse patchAddress(@Nonnull Cart cartWithPaymentsExpansion,
                                               @Nonnull String paypalPlusPaymentId) {
         try {
             Optional<String> paymentIdOpt = getCtpPaymentId(cartWithPaymentsExpansion, paypalPlusPaymentId);
             return paymentIdOpt.map(paymentId -> {
                 Payment paypalPlusPayment = new Payment().setId(paypalPlusPaymentId);
-                ShippingAddress shippingAddress = shippingAddressMapper.ctpAddressToPaypalPlusAddress(cartWithPaymentsExpansion.getShippingAddress());
-                Patch replace = new Patch("add", "/transactions/0/item_list/shipping_address").setValue(shippingAddress);
-                AddInterfaceInteraction addInterfaceInteractionAction = createAddInterfaceInteractionAction(replace, REQUEST);
+                ImmutableList.Builder<Patch> builder = ImmutableList.builder();
+
+                io.sphere.sdk.models.Address shippingAddress = cartWithPaymentsExpansion.getShippingAddress();
+                Patch patchShippingAddress = new Patch("add", "/transactions/0/item_list/shipping_address").setValue(addressMapper.ctpAddressToPaypalPlusShippingAddress(shippingAddress));
+                builder.add(patchShippingAddress);
+
+                // billing address is not mandatory
+                io.sphere.sdk.models.Address billingAddress = cartWithPaymentsExpansion.getBillingAddress();
+                if (billingAddress != null) {
+                    Patch patchBillingAddress = new Patch("add", "/payer/payer_info")
+                            .setValue(addressMapper.ctpAddressToPaypalPlusPayerInfo(billingAddress));
+                    builder.add(patchBillingAddress);
+                }
+                AddInterfaceInteraction addInterfaceInteractionAction = createAddInterfaceInteractionAction(patchShippingAddress, REQUEST);
                 CompletionStage<PaymentHandleResponse> patchCS = ctpFacade.getPaymentService()
                         // 1. add patch request to the ctp payment interface interaction
                         .updatePayment(paymentId, Collections.singletonList(addInterfaceInteractionAction))
                         .thenCompose(payment -> paypalPlusFacade.getPaymentService()
                                 // 2. patch payment on paypal
-                                .patch(paypalPlusPayment, replace)
+                                .patch(paypalPlusPayment, builder.build())
```

Comment:
I'd not recommend to use the `ImmutableBuilder` here, it doesn't give any benefits in this case (immutability, shrunk capacity etc). 

Having only 2 potential entries in the list it is enough to use any simple default Java list implementation (e.g. `ArrayList`)

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/51/commits/5288d7243d7e90fc0bfc661e5fd710dfa327b84e